docs/core/deploying/index.md

```diff
@@ -219,58 +218,192 @@ A complete sample `project.json` file appears later in this section.
+   },
+   ```
+
+3. If you haven't already, download the NuGet package containing the third-party dependency to your system. To make the dependency available to your app, execute the `dotnet restore` command after adding the dependency. Because the dependency is resolved out of the local NuGet cache at publish time, it must be available on your system.
+
+The following is the complete project.json file for this project:
+
+```json
+{
+  "version": "1.0.0-*",
+  "buildOptions": {
+    "debugType": "portable",
+    "emitEntryPoint": true
+  },
+  "dependencies": {
+    "Microsoft.NETCore.App": "1.0.0",
+    "Newtonsoft.Json": "8.0.3"
+  },
+  "frameworks": {
+    "netcoreapp1.0": {
+      "imports": "dnxcore50"
+    }
+  },
+  "runtimes": {
+    "win10-x64": {},
+    "osx.10.10-x64": {}
+  }
+}
+```
 
 When you deploy your application, any third-party dependencies used in your app are also contained with your application files. Third-party libraries do not already have to be present on the system on which the app is running.
 
 Note that you can only deploy a self-contained app with a third-party library to platforms supported by that library.
 
+### Deploying a Self-Contained App with a Smaller Footprint ###
+
+If the availability of adequate storage space on target systems is likely to be an issue, you can reduce the overall footprint of your app by excluding some system components. To do this, you explicitly define the .NET Core components that your app includes in your project.json file.
+
+To create a self-contained app with a smaller footprint, start by following the first two steps for creating a self-contained app. Once you've run the `dotnet new` command and added the C# source code to your app, do the following:
+
+1. Open the `project.json` file and replace the `frameworks` section with the following:
+
+    ```json
+    "frameworks": {
+      "netstandard1.6": { }
+    }
+    ```
+This does two things:
+
+    * It indicates that, instead of using the entire `netcoreapp1.0` framework, which includes .NET Core CLR, the .NET Core Library, and a number of other system components, our app uses only the .NET Standard Library.
+
+    * By removing the `"type": "platform"` attribute, it indicates that the framework is provided as a set of components local to our app, rather than as a system-wide platform package.
+
+2. Replace the `dependencies` section with the following:
+
+    ```json
+    "dependencies": {
+      "NETStandard.Library": "1.6.0",
+      "Microsoft.NETCore.Runtime.CoreCLR": "1.0.2",
+      "Microsoft.NETCore.DotNetHostPolicy":  "1.0.1"
+    },
+    ```
+   This defines the system components used by our app. The system components packaged with our app include the .NET Standard Library, the .NET Core runtime, and the .NET Core host. This produces a self-contained app with a smaller footprint than if you had simply modified the `frameworks` section to indicate that your app is providing the entire `netcoreapp1.0` framework.
```

Comment:
I'm wondering if this can be written slightly differently. I would keep to the "package" term and not involve the target ("netcoreapp1.0") in this. When you modify the framework section in the previous example, you add a dependency on a given meta-package. It is the contents of this meta-package that are published out.